File: minNumInRotatedSortedArr.js
function minNumInRotatedSortedArr(nums) {
	let l = 0, r = nums.length - 1
	let min = nums[0]

	while (l <= r) {
		if (nums[l] < nums[r]) {
			min = Math.min(min, nums[l])
			break
		}

		let mid = Math.floor((l + r) / 2)
		min = Math.min(min, nums[mid])

		if (nums[l] <= nums[mid]) {
			l = mid + 1
		} else {
			r = mid - 1
		}
	}

	return min
}

console.log(minNumInRotatedSortedArr([6, 7, 8, 9, 2, 3]))

File: minNumInRotatedSortedArr.java
class Solution {
	public int minNumInRotatedSortedArr(int[] nums) {
		int l = 0, r = nums.length - 1;
		int res = nums[0];

		while (l <= r) {
			if (nums[l] < nums[r]) {
				res = Math.min(res, nums[l]);
				break;
			}

			int m = (l + r) / 2;
			res = Math.min(res, nums[m]);

			if (nums[l] <= nums[m]) 
				l = m + 1;
			else 
				r = m - 1;
 		}

 		return res;
	}

	public static void main(String[] args) {
		int[] nums = {6, 7, 8, 9, 2, 3};
		Solution s = new Solution();
		System.out.println(s.minNumInRotatedSortedArr(nums));
	}
}